### PR TITLE
Use gradio-api-server for telemetry

### PR DESCRIPTION
**Pull request by dawoodkhan82**
# Description
Sends requests for telemtry to gradio-api-server  (https://github.com/gradio-app/gradio-api-server/pull/8)

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Closes: # (issue)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by abidlabs** [comment]
Everything looks good to me @dawoodkhan82! 

I've merged the gradio api server PR and restarted the API server, so can you double check that the analytics are going through from this branch before we merge in?

**Comment by dawoodkhan82** [comment]
Doesn't seem like its working. Will take a look tomorrow and try to debug it locally. Elastic search must not like how I'm passing the `request.form`.

**Comment by abidlabs** [comment]
Yeah maybe it needs to be converted to a dictionary explicitly

**Comment by dawoodkhan82** [comment]
@abidlabs What do you suggest I pass [here](https://github.com/gradio-app/gradio/pull/3488/files#diff-0dd827ecd796f1d09e48052f6f80c1c9cf519eb940595994c65744ca40b0ae6aR168) instead of `blocks.theme`?

**Comment by abidlabs** [comment]
I'd suggest adding a `.name` attribute to the `ThemeClass` class. This name is set by the 6 predefined themes to be the class name. And in the `ThemeClass.from_hub()` method, it should set it to the `repo_name`. 

(cc @freddyaboulton)

**Comment by dawoodkhan82** [comment]
This wouldn't track local custom themes tho right? @abidlabs

**Comment by freddyaboulton** [comment]
Good idea @abidlabs . That wouldn't capture cases where someone does `gr.themes.Monochrome().set(.....)`. I suggest we also add a flag for `is_custom_theme`. We can define this as whether or not the theme is equal to any of the pre-defined themes, e.g. `not any(theme.to_dict() == built_in_theme.to_dict() for built_in_theme in [gr.themes.Monochrome(), ....])`

**Comment by abidlabs** [comment]
Correct, it would not. We could also add an attribute for that but maybe it's not necessary at this point (?)

**Comment by abidlabs** [comment]
> We can define this as whether or not the theme is equal to any of the pre-defined themes, e.g. not any(theme.to_dict() == built_in_theme.to_dict() for built_in_theme in [gr.themes.Monochrome(), ....])

Cool suggestion

**Comment by dawoodkhan82** [comment]
ready for re-review btw

**Comment by dawoodkhan82** [comment]
Thanks for adding `name` to themes @freddyaboulton!

File: gradio/themes/base.py [hunk 1]
```diff
@@ -326,6 +326,7 @@ class Base(ThemeClass):
     def __init__(
         self,
         *,
+        name: str = "base",
```
Comment (abidlabs) [inline gradio/themes/base.py:329]: Looks like an unused parameter?

**Comment by abidlabs** [comment]
LGTM @dawoodkhan82! Thanks for making the fix and testing it. One nit above, but otherwise, it the analytics are showing up for you on Kibana when you test this locally, then we should be good to go!

**Review by freddyaboulton — APPROVED** [review]
Very cool @dawoodkhan82 ! Looks good to me. Thanks for adding all the theme related analytics!

**Pull request merged**